src/lib/charm/openstack/keystone_saml_mellon.py

```diff
@@ -101,17 +83,24 @@ def sp_private_key_file(self):
     def sp_location_config(self):
         return SP_LOCATION_CONFIG
 
+    def get_fid_data(self):
+        fid_sp = endpoint_from_flag(KEYSTONE_FID_ENDPOINT)
+        if fid_sp:
+            return fid_sp.all_joined_units.received
+        else:
+            return {}
+
     @property
     def keystone_host(self):
-        return unitdata.kv().get('hostname')
+        return self.get_fid_data().get("hostname")
```

Comment:
Do these properties belong here and are they necessary anymore now that you are exposing them on the interface directly?

Reply:
Good point. I have removed any unit data and updated the template to use the interface data.